### ISSUE TITLE
Multiple inheritence does not work correctly with relative operators

### ISSUE THREAD

**Issue by colinjneville** [andymeneely/squib#244]
When a layout makes a relative adjustment to a key that does not appear in the last layout specified in `extends`, the value for that key is left as the string value including the operator (e.g. `'+= 100'`).
For example, the example in the [documentation for multiple inheritence](http://squib.readthedocs.io/en/v0.14.0/layouts.html#yes-extends-has-multiple-inheritance) will fail because `x` is a string:
```yaml
socrates:
  x: 100
plato:
  y: 200
aristotle:
  extends:
    - socrates
    - plato
  x: += 50    # evaluates to 250 from plato
```
```ruby
require 'squib'

Squib::Deck.new(cards: 1, layout: 'test.yml') do
  circle radius: 50, layout: :aristotle
  
  save_png prefix: 'test_'
end
```
```
lib/ruby/2.4.0/forwardable.rb:229:in `arc': no implicit conversion to float from string (TypeError)
```
I'm new to Ruby, so I may be reading something wrong, but it looks like [recurse_extends](https://github.com/andymeneely/squib/blob/498a8224e70bb0180dd7c460b867037027909189/lib/squib/layout_parser.rb#L43) merges the current layout with each parent separately (so `aristotle/socrates['x']` is `150`, `aristotle/plato['x']` is `'+= 50'`), then those are all merged with the later parents taking priority when multiple values exist for the same key. Since `aristotle/plato['x']` technically has a value, it overwrites the expected value, so `aristotle['x'] == '+= 50'`.

**Comment by andymeneely** [comment]
Yeah you're right - I remember fixing this somewhere back along the way. Wonder what happened. 

You can also see what the layout computes to with a `puts layout` in there. In this case it comes out to 

```
{"socrates"=>{"x"=>100}, "plato"=>{"y"=>200}, "aristotle"=>{"extends"=>["socrates", "plato"], "x"=>"+= 50", "y"=>200}}
```
Good catch. Looking into it.

**Comment by andymeneely** [comment]
Turns out I never tested this case. I think I've handled it. The docs were also wrong, so I fixed that too. The new docs will be correct once I put out the new release. Expect this fix in v0.14.2
